feat(events-v2) Enable clicking on graph markers

markstory opened the pull request:
Clicking on a graph marker will now take you to an event in the selected timeslice. This requires an extra request to find an event in the slice. I've opted to use the `/latest` endpoint so that I don't have to deal with getting more a single result back, and we bias towards newer data in other event views.

Refs SEN-733

billyvg commented on src/sentry/static/sentry/app/components/charts/utils.jsx:
```diff
@@ -56,6 +56,28 @@ export function getInterval(datetimeObj, highFidelity = false) {
+ * @param {String} interval The interval to convert.
+ * @return {Integer}
+ */
+export function intervalToMilliseconds(interval) {
```
We have this to hours: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils.jsx#L246

markstory replied:
I can move this to live with its friend. I wonder if both shouldn't live in utils/dates.jsx?

billyvg replied:
I personally find the util functions hard to discover when they are grouped together in the same files, but moving them into `utils/dates` is an improvement 👍

billyvg commented on src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx:
```diff
@@ -49,6 +59,55 @@ const getCurrentEventMarker = currentEvent => {
+      query,
+    });
+  } catch (e) {
+    // Do nothing, user could have clicked on a blank space.
```
Thoughts (in general) about `console.error` in catch blocks? Sometimes they can end up swallowing errors in development and testing too.

markstory replied:
I think it makes sense when the failure is not expected to happen. I'm certain that users will click in blank spaces causing 404. I don't want to be notified in sentry when that happens though.

billyvg replied:
`console.error` won't log to sentry, I'm mainly thinking about during dev/tests (e.g. in tests if you don't mock this endpoint, the test will pass, maybe not applicable here, but in other places it's more relevant)

markstory commented:
I've moved the interval parsing functions together and added the pin icon to the current event marker. The image is a bit wacky as echarts inverts our image when it positions it on the chart.

![Screen Shot 2019-06-14 at 11 08 22 AM](https://user-images.githubusercontent.com/24086/59518981-e7657200-8e94-11e9-8f5c-8a26f75e51d9.png)

lynnagara commented:
Don't we know for sure that at least one event exists at a timestamp if a clickable marker has been rendered there? i.e. Could range just be 0 or 1 millisecond?

markstory commented:
@lynnagara We can't use narrow ranges. As an example when you're looking at 14 days of data, each 'slice' in the graph represents 30 minutes of time, and we don't know where in the 30 minutes the event you're looking for is.

lynnagara commented on src/sentry/static/sentry/app/views/organizationEvents/utils/eventsRequest.jsx:
```diff
@@ -46,6 +46,7 @@ class EventsRequest extends React.PureComponent {
      * Absolute start date for query
      */
     start: PropTypes.instanceOf(Date),
+
```
nit: no need to touch this file
